### PR TITLE
Remove misleading docs around eviction and image garbage collection

### LINKED ISSUE #43303: Misleading documentation around eviction and image garbage collection
We say that the ImageGCHighThresholdPercent/ImageGCLowThresholdPercent config option is deprecated for evictionHard settings but this is not true.

https://kubernetes.io/docs/concepts/scheduling-eviction/node-pressure-eviction/#eviction-signals

ImageGC uses `image-gc-high-threshold-percent` and `image-gc-low-threshold-percent` for Garbage collection so these flags are not [deprecated](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet.go#L421)

We should update our documentation to reflect this.

### PR DESCRIPTION
**Pull request by Affan-7**
This closes #43303

Deploy preview: https://deploy-preview-43336--kubernetes-io-main-staging.netlify.app/docs/concepts/scheduling-eviction/node-pressure-eviction/#eviction-signals

### CODE REVIEW

File: content/en/docs/concepts/scheduling-eviction/node-pressure-eviction.md [hunk 1]
```diff
@@ -105,13 +105,11 @@ does not support other configurations.
 
 Some kubelet garbage collection features are deprecated in favor of eviction:
```
Comment (kannon92) [inline content/en/docs/concepts/scheduling-eviction/node-pressure-eviction.md:106]: So just to be clear,

`--eviction-hard` and `--eviction-soft` are still used to trigger garbage collection.  

We would trigger garbage collection if these signals are passed.  

And we also still use `--image-gc-high-threshold` and `--image-gc-low-threshold` for image garbage collection policies.

**Review by sftim — COMMENTED** [review]
/approve

**Pull request merged**